lib/genesis/genesis.go

```diff
@@ -46,8 +50,157 @@ type TelemetryEndpoint struct {
+	PhragmenElection    *phragmenElection    `json:"PhragmenElection"`
+	Instance1Membership *instance1Membership `json:"Instance1Membership"`
+	Contracts           *contracts           `json:"Contracts"`
+	Society             *society             `json:"Society"`
+	Indices             *indices             `json:"indices"`
+	ImOnline            *imOnline            `json:"imOnline"`
+	AuthorityDiscovery  *authorityDiscovery  `json:"authorityDiscovery"`
+	Vesting             *vesting             `json:"vesting"`
+	NominationPools     *nominationPools     `json:"nominationPools"`
+	Configuration       *configuration       `json:"configuration"`
+	Paras               *paras               `json:"paras"`
+	Hrmp                *hrmp                `json:"hrmp"`
+	Registrar           *registrar           `json:"registrar"`
+	XcmPallet           *xcmPallet           `json:"xcmPallet"`
+}
+
+// System is ...
+type System struct {
+	Code string `json:"code"`
+}
+
+type babe struct {
+	Authorities []types.AuthorityAsAddress `json:"authorities"`
+	EpochConfig *epochConfig               `json:"epochConfig"`
+}
+
+type epochConfig struct {
+	C            []int  `json:"c"`
+	AllowedSlots string `json:"allowed_slots"`
+}
+
+type grandpa struct {
+	Authorities []types.AuthorityAsAddress `json:"authorities"`
+}
+
+type balances struct {
+	Balances []balancesFields `json:"balances"`
+}
+
+type balancesFields struct {
+	AccountID string
+	Balance   float64
+}
+
+type sudo struct {
+	Key string `json:"key"`
+}
+
+type indices struct {
+	Indices []interface{} `json:"indices"`
+}
+
+type imOnline struct {
+	Keys []interface{} `json:"keys"`
+}
+
+type authorityDiscovery struct {
+	Keys []interface{} `json:"keys"`
+}
+
+type vesting struct {
+	Vesting []interface{} `json:"vesting"`
+}
+
+type nominationPools struct {
+	MinJoinBond       int64 `json:"minJoinBond"`
+	MinCreateBond     int64 `json:"minCreateBond"`
+	MaxPools          int   `json:"maxPools"`
+	MaxMembersPerPool int   `json:"maxMembersPerPool"`
+	MaxMembers        int   `json:"maxMembers"`
+}
+
+type configuration struct {
+	Config config `json:"config"`
+}
+
+type config struct {
+	MaxCodeSize                           int         `json:"max_code_size"`
+	MaxHeadDataSize                       int         `json:"max_head_data_size"`
+	MaxUpwardQueueCount                   int         `json:"max_upward_queue_count"`
+	MaxUpwardQueueSize                    int         `json:"max_upward_queue_size"`
+	MaxUpwardMessageSize                  int         `json:"max_upward_message_size"`
+	MaxUpwardMessageNumPerCandidate       int         `json:"max_upward_message_num_per_candidate"`
+	HrmpMaxMessageNumPerCandidate         int         `json:"hrmp_max_message_num_per_candidate"`
+	ValidationUpgradeCooldown             int         `json:"validation_upgrade_cooldown"`
+	ValidationUpgradeDelay                int         `json:"validation_upgrade_delay"`
```

Comment:
Why pointer for uint ?

Reply:
So that if they are left unset in the JSON, the result is `nil` instead of `0`.
I would guess the default value should not always be 0 if it's left unset for all integer fields 🤔